In [ ]:
#loading the data for accessing
data_file = 'KAG_Conversion_Data.csv'
import numpy as np

data = np.genfromtxt(data_file, delimiter=",", skip_header=1, dtype= int)
data_str=np.genfromtxt(data_file, delimiter=',',skip_header=1,dtype=str)
data_fl=np.genfromtxt(data_file,delimiter=',',skip_header=1,dtype=float)

print("\nData: \n\n", data)


Data: 

 [[ 708746     916  103916 ...      -1       2       1]
 [ 708749     916  103917 ...      -1       2       0]
 [ 708771     916  103920 ...       0       1       0]
 ...
 [1314412    1178  179979 ...      -1       2       0]
 [1314414    1178  179981 ...      -1       8       2]
 [1314415    1178  179982 ...      -1       5       2]]


In [ ]:
data_str

array([['708746', '916', '103916', ..., '1.429999948', '2', '1'],
       ['708749', '916', '103917', ..., '1.820000023', '2', '0'],
       ['708771', '916', '103920', ..., '0', '1', '0'],
       ...,
       ['1314412', '1178', '179979', ..., '40.28999949', '2', '0'],
       ['1314414', '1178', '179981', ..., '198.7100005', '8', '2'],
       ['1314415', '1178', '179982', ..., '165.6099987', '5', '2']],
      dtype='<U11')

In [ ]:
#identify unique campaigns
ad_campaigns = np.array(data[:,1])
print(ad_campaigns)

[ 916  916  916 ... 1178 1178 1178]


In [ ]:
#pulling out unique ad_campaign
np.unique(ad_campaigns)

array([ 916,  936, 1178])

In [ ]:
data[:,7]

array([  1,   2,   0, ...,  28, 135, 114])

In [ ]:
#determine how many times each capaign was run
run_frequency = np.unique(ad_campaigns,return_counts=True)
print(run_frequency)

(array([ 916,  936, 1178]), array([ 54, 464, 625]))


The result means ad campaign 916 was runned 54times, ad campaign 936 was runned 464times and ad campaign 1178 was runned 625times


In [ ]:
#identify the target age group
age_group = np.array(data_str[:,3])
print(age_group)



['30-34' '30-34' '30-34' ... '45-49' '45-49' '45-49']


In [ ]:
np.unique(age_group)

array(['30-34', '35-39', '40-44', '45-49'], dtype='<U11')

The above Age groups were targeted 

In [ ]:
#determine the average, min and max amount spent on these ads
#we start by pulling out the amount_spend column(index 8)
amount_spend = data_fl[:,8]
print(amount_spend)
#average_amount = np.average(amount_spend)
#print(average_amount)


[  1.42999995   1.82000002   0.         ...  40.28999949 198.7100005
 165.6099987 ]


In [ ]:
np.mean(amount_spend)

51.36065613141295

In [ ]:
np.max(amount_spend)

639.9499981

In [ ]:
np.min(amount_spend)

0.0

In [ ]:
#pull out the number of clicks column and identify the maximum number of clicks
clicks = np.array(data[:,7])
print(clicks)
max_clicks = clicks.max()
print(max_clicks)

[  1   2   0 ...  28 135 114]
421


In [76]:
#identify the id that corresponds to the max number of clicks
np.where(clicks==max_clicks)

(array([860]),)

The maximum number of clicks has index 860

In [78]:
#identify the id that matches with the max number of clicks
data[: ,0][data[:,7]==max_clicks]


array([1121814])

In [79]:
#number ofpeople who bought the product after seeing the max clicks
data[:,10][data[:,7]== max_clicks]

array([13])

In [82]:
#find the max number of purchase
max_bought = np.array(data[:,10])
print(max_bought)
max_bought_per_click = max_bought.max()
print(max_bought_per_click)

[1 0 0 ... 0 2 2]
21


the ad for which the purchases are max is not the one with the max clicks


In [83]:
data[:][data[:,10]==21]

array([[1121104,    1178,  144533,      -1,      -1,      16, 2080666,
            202,      -1,      40,      21]])

The above are the details of the ad with max number of purchases